### PR TITLE
OLS-833: Bump-up Ruff to version 0.5.2

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Bump-up Ruff to version 0.5.2

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [x] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)

## Related Tickets & Documents

- Related Issue #[OLS-833](https://issues.redhat.com//browse/OLS-833)

### CODE REVIEW

File: tests/e2e/utils/decorators.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def wrapper(*args, **kwargs):
                     return test_function(*args, **kwargs)
 
                 except Exception as e:
-                    if type(e) != on_error:
+                    if not isinstance(e, on_error):
```
Comment (raptorsun) [inline tests/e2e/utils/decorators.py:37]: aren't they working the same way?

Comment (tisnik) [reply tests/e2e/utils/decorators.py:37]: no, unlike a direct type comparison, `isinstance` will also check if an object
is an instance of a class or a subclass thereof. New Ruff is taking care of it (I have not found it personally ;)

**Pull request merged**